### PR TITLE
Ei 198 add get application endpoint

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives/Application/Queries/GetApplication/GetApplicationQuery.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using System;
+using MediatR;
+
+namespace SFA.DAS.EmployerIncentives.Application.Queries.GetApplication
+{
+    public class GetApplicationQuery : IRequest<GetApplicationResult>
+    {
+        public long AccountId { get ; set ; }
+        
+        public Guid ApplicationId { get; set; }
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives/Application/Queries/GetApplication/GetApplicationQuery.cs:10]: i was asked to use `IncentiveApplicationId` just FYI

File: src/SFA.DAS.EmployerIncentives/InnerApi/Requests/GetApplicationRequest.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+using System;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.EmployerIncentives.InnerApi.Requests
+{
+    public class GetApplicationRequest : IGetApiRequest
```
Comment (dashton82) [inline src/SFA.DAS.EmployerIncentives/InnerApi/Requests/GetApplicationRequest.cs:6]: Missing a test for this 👀

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/ApplicationController.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            return new IncentiveApplicationDto
+            {
+                Apprenticeships = application.Apprenticeships.Select(x => MapToDto(x))
+            };
+        }
+
+        private IncentiveApplicationApprenticeshipDto MapToDto(IncentiveApplicationApprenticeship apprenticeship)
+        {
+            return new IncentiveApplicationApprenticeshipDto
```
Comment (dashton82) [inline src/SFA.DAS.EmployerIncentives.Api/Controllers/ApplicationController.cs:60]: Can these private methods not exist on the Model instead using an implicit operator ? Will keep the controller a bit cleaner

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/ApplicationController.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        private readonly ILogger<ApplicationController> _logger;
+
+        public ApplicationController(IMediator mediator, ILogger<ApplicationController> logger)
+        {
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        [Route("/accounts/{accountId}/applications/{applicationId}")]
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Api/Controllers/ApplicationController.cs:26]: hmm i thought `/accounts/{accountId}/` is being automatically added somewhere :/

**Pull request merged**